src/server/rpc/procedures/utils/index.js

```diff
@@ -42,6 +42,18 @@ const jsonToSnapList = inputJson => {
     return keyVals;
 };
 
+// converts snap friendly  key value array to json object
+// parses 1 level only!
+const kvListToJson = snapArray => {
+    let parsedObj = {};
+    if (snapArray.length === 0) return {};
+    snapArray.forEach(kv => {
+        let [k,v] = kv;
+        if (parsedObj[k] !== undefined) throw 'duplicate key in input list found';
```

Comment:
Are you sure this is the best behavior? It might make more sense to not fail here...